### DISCUSSION TITLE
Unable to add element in ArraySchema at specified position

### DISCUSSION THREAD

**Discussion by zack1991** [colyseus/colyseus#632] [General]
Hi,

When i am trying to add element in ArraySchema at specified index using splice it is not getting added.

```
const arrayItems = new ArraySchema(1, 2, 4);
arrayItems.splice(2, 0, 3);
printLog(`arrayItems.length => ${arrayItems.length}`);

Output :- 
arrayItems.length => 3
```

**Comment by endel** [comment]
Hi @zack1991, indeed [the third argument of the `splice()` method](https://github.com/colyseus/schema/blob/aef7a9a9f1866be14745c35884830df36559d7c4/src/types/ArraySchema.ts#L316) hasn't been implemented. This has been reported in the past and the status is still the same https://github.com/colyseus/schema/issues/101#issuecomment-764663202. 

The list of issues `ArraySchema` has keeps increasing, the way arrays are synchronized must be revisited and almost re-written from scratch, I'm afraid the "proper" implementation will come only next year, although hotfixes can be applied in the current implementation in the meantime https://github.com/colyseus/colyseus/wiki/Public-Roadmap#colyseus-framework
